source/material_model/morency_doin.cc

```diff
@@ -93,7 +93,7 @@ namespace aspect
             for (unsigned int j=0; j < volume_fractions.size(); ++j)
               np += volume_fractions[j] * nps[j];
 
-            const double tauy = tau_0 + gamma*reference_density()*g*depth;
+            const double tauy = tau_0 + gamma*densities[0]*g*depth;
```

Comment:
Yes, that seems like the right decision then. Thanks for thinking this issue through.